typing/uniqueness_analysis.ml

```diff
@@ -820,6 +832,8 @@ end = struct
 
   let variant_field s t = child (Projection.Variant_field s) t
 
+  let array_index gf i t = modal_child gf (Projection.Array_index i) t
```

Comment:
Maybe better to let this function take `mut` instead of `gf`. Because currently the modalities is decided by solely `mut`, so the `gf` would have only two possible values here.